azure-pipelines/templates/matrix-job.yml

```diff
@@ -0,0 +1,47 @@
+parameters:
+  configuration: ['Debug', 'Release']
+  sdl2_static: ['ON', 'OFF']
+  build_testing:  ['ON', 'OFF']
+  build_shared_libs: ['ON', 'OFF']
+  build_examples: ['ON', 'OFF']
+  build_snapshot: ['ON', 'OFF']
+  enable_code_coverage: ['ON', 'OFF']
+  vmImageName: ''
+  triplet: ''
+  jobName: 'build_matrix'
+  arch: ''
+  cmake_ignore_path: ''
+  
+jobs:
+- job: ${{ parameters.jobName }}
+  #displayName: ${{ parameters.jobDisplayName }}
+  strategy:
+    matrix:
+      ${{ each configuration in parameters.configuration }}:
+        ${{ each sdl2_static in parameters.sdl2_static}}:
+          ${{ each build_testing in parameters.build_testing }}:
+            ${{ each build_shared_libs in parameters.build_shared_libs }}:
+              ${{ each build_examples in parameters.build_examples }}:
+                ${{ each build_snapshot in parameters.build_snapshot}}:
+                  ${{ each enable_code_coverage in parameters.enable_code_coverage}}:
+                    ${{ format('{0}-{1}-{2}-{3}-{4}-{5}-{6}', configuration, sdl2_static, build_testing, build_shared_libs, build_examples, build_snapshot, enable_code_coverage) }}:
```

Comment:
a better name should be provided, is quite unreadable...

Reply:
i won't use anymore, just done for fun and completeness... so at the moment can be kept in that "coded" fashion.